ui/src/dms/Dm.tsx

```diff
@@ -69,43 +53,51 @@ export default function Dm() {
-                  to="/"
-                  className={cn(
-                    'default-focus ellipsis inline-flex appearance-none items-center pr-2 text-lg font-bold text-gray-800 sm:text-base sm:font-semibold'
-                  )}
-                  aria-label={isMobile ? 'Open Messages Menu' : undefined}
-                >
-                  {isMobile ? (
-                    <CaretLeft16Icon className="mr-2 h-4 w-4 shrink-0 text-gray-400" />
-                  ) : null}
-                  <div className="mr-3 flex h-6 w-6 shrink-0 items-center justify-center rounded bg-gray-100 text-center">
-                    <Avatar size="xs" ship={ship} />
+            <div className="flex items-center justify-between border-b-2 border-gray-50 bg-white py-2 pl-2 pr-4">
+              <BackButton
+                to="/"
+                className={cn(
+                  'default-focus ellipsis w-max-sm inline-flex h-10 appearance-none items-center justify-center space-x-2 rounded p-2'
+                )}
+                aria-label="Open Messages Menu"
+              >
+                {isMobile ? (
+                  <div className="flex h-6 w-6 items-center justify-center">
+                    <CaretLeft16Icon className="h-5 w-5 shrink-0 text-gray-600" />
                   </div>
-                  <span className="ellipsis text-gray-200 line-clamp-1">
-                    {contact?.nickname ? (
-                      <>
-                        <span className="text-gray-800">
-                          {contact.nickname}
-                        </span>
-                        <span className="ml-2 text-gray-400">{ship}</span>
-                      </>
-                    ) : (
-                      <span className="text-gray-800">{ship}</span>
-                    )}
-                  </span>
-                </BackButton>
-              </BackLink>
-              <div className="flex shrink-0 flex-row items-center space-x-3 self-end">
+                ) : null}
+                <div className="flex h-6 w-6 shrink-0 items-center justify-center rounded text-center">
+                  <Avatar size="xs" ship={ship} />
+                </div>
+                <div className="flex w-full flex-col justify-center">
+                  {contact?.nickname ? (
+                    <>
+                      <span
+                        className={cn(
+                          'ellipsis text-sm font-bold line-clamp-1 sm:font-semibold'
+                        )}
+                      >
+                        {contact.nickname}
+                      </span>
+                      <span className="w-full break-all text-sm text-gray-400 line-clamp-1">
+                        {ship}
+                      </span>
+                    </>
+                  ) : (
+                    <span className="text-sm font-bold text-gray-800 sm:font-semibold">
+                      {ship}
+                    </span>
+                  )}
```

Comment:
you'll have to import shipname, but it should probably be:

```suggestion
                      <ShipName full name={ship} className="w-full break-all text-sm text-gray-400 line-clamp-1" />
                    </>
                  ) : (
                    <ShipName full name={ship} className="text-sm font-bold text-gray-800 sm:font-semibold" />
                  )}
```

Reply:
Thanks!